### DISCUSSION TITLE
After updating from v2.5.12 to v2.6.0,

### DISCUSSION THREAD

**Discussion by fade2gray** [mark-wiemer/ahkpp#30] [Q&A] [answered]
I notice that after updating from v2.5.12 to v2.6.0, the older version remains in the `.vscode\extensions` directory for an indefinite period.  I believe it happens when other extensions are updated, but why does this happen at all?

**Comment by mark-wiemer** [comment, marked as answer]
I'm not seeing the older version in my filesystem. I would guess it's just an optimization in VS Code to remove old extensions in batches instead of individually

**Comment by fade2gray** [comment]
![Capture](https://user-images.githubusercontent.com/1332009/105214839-7003c400-5b48-11eb-8704-7e89ac8054fc.PNG)

**Reply by mark-wiemer** [reply]
Oh, I didn't mean to doubt that you were seeing this in your filesystem! What I meant is that it doesn't seem like a problem. I saw extensions I had recently uninstalled in that folder, but they went away once I restarted VS Code.

**Reply by fade2gray** [reply]
Yes, the earlier version is removed from the extension directory only after VSC is restarted, but not when clicking `Reload Required`.

This tripped me up when composing a script that parses the directory for all extensions for a `package.json` containing a particular keyword and subsequently obtaining other data from that file. When two versions of the same extension were found, because the earlier version comes alphabetically before the later, my script was getting the incorrect data and erroring out. Fixed by ensuring that script parsed for a list of all versions of the required extension and only working with that which came alphabetically last.
